### DISCUSSION TITLE
Add healthcheck instructions to example Dockerfile in the documentation

### DISCUSSION THREAD

**Discussion by nikstuckenbrock** [fastapi/fastapi#5035] [Questions]
I recently read a recommendation to add the `HEALTHCHECK` command to a Dockerfile. I was wondering whether it would make sense to include a default `/health` endpoint in FastAPI and use this for the Docker `HEALTHCHECK`.
Looking forward for some feedback on this idea.

I found the recommendation in the [@bridgecrewio/checkov](bridgecrewio/checkov) [documentation](https://docs.bridgecrew.io/docs/ensure-that-healthcheck-instructions-have-been-added-to-container-images):

> We recommend that you add the HEALTHCHECK instruction to your Docker container images to ensure that health checks are executed against running containers.
>
> An important security control is that of availability. Adding the HEALTHCHECK instruction to your container image ensures that the Docker engine periodically checks the running container instances against that instruction to ensure that containers are still operational.
>
> Based on the results of the health check, the Docker engine could terminate containers which are not responding correctly, and instantiate new ones.

**Comment by odiseo0** [comment]
It would make sense, yes. You can also use the root endpoint for the healthcheck `/`

**Reply by zoliknemet** [reply]
problem with root endpont `/` is that returns `404` and response body `{"detail":"Not Found"}`
in this case if you want quick soluction,  `/docs` would be better :)

When I am working on /healthcheck I usually do custom method which for example makes `SELECT 1; ` on database, just to be sure my connections are fine. You can also do other checks on dependencies that you are using.

**Reply by nikstuckenbrock** [reply]
Thanks for your feedback!
@odiseo0: I like the idea, but I see the same problem as @zoliknemet mentioned.
@zoliknemet: Wouldn't something like a `SELECT` be the healthcheck for the database container itself so you don't need to check it in the API container?

**Reply by zoliknemet** [reply]
@nikstuckenbrock yeah it would, but when you are checking `SELECT` from app part, you also check your database connection pool, not just database itself - database can be up and running, but your app disconnected from it (out of some reason)
It is just a thing I do...

**Reply by nikstuckenbrock** [reply]
Thanks @zoliknemet!
I think I will create an issue and a PR then to at least include a basic `HEALTHCHECK` in the docs by default.
I really appreciate your time and help.
